awx/main/analytics/collectors.py

```diff
@@ -78,16 +104,26 @@ def events_slicing(key, since, until):
     lower = since or last_gather or horizon
     if not since and last_entries.get(key):
         lower = horizon
-    pk_values = models.JobEvent.objects.filter(created__gte=lower, created__lte=until).aggregate(Min('pk'), Max('pk'))
+    partition_epoch = get_event_partition_epoch()
 
-    previous_pk = pk_values['pk__min'] - 1 if pk_values['pk__min'] is not None else 0
-    if not since and last_entries.get(key):
-        previous_pk = max(last_entries[key], previous_pk)
-    final_pk = pk_values['pk__max'] or 0
+    # Stage 1: Process any job events in unpartitioned job event table
```

Comment:
@jbradberry this is still a work in progress. We still need to get some data on the (1) only collect data by scooping up partitions approach vs the (2) get partitions when going up to the horizon, otherwise collect by pk / modified field.

This is written to follow the second approach, but I'd actually like to have it so that we can toggle between both options to make testing at scale easier.